File: src/main/java/com/ggardet/hospital/internal/repository/TreatmentRepository.java
package com.ggardet.hospital.internal.repository;

import com.ggardet.hospital.internal.entity.Treatment;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface TreatmentRepository extends JpaRepository<Treatment, UUID> {
    List<Treatment> findAllByHospitalId(final UUID hospitalId);
}

File: src/main/kotlin/com/ggardet/hospital/internal/repository/TreatmentRepository.kt
package com.ggardet.hospital.internal.repository

import com.ggardet.hospital.internal.entity.Treatment
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface TreatmentRepository : JpaRepository<Treatment, UUID> {
    fun findAllByHospitalId(hospitalId: UUID): List<Treatment>
}